packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -275,13 +292,30 @@ class BuildInfo {
       if (codeSizeDirectory != null)
         'CODE_SIZE_DIRECTORY': codeSizeDirectory!,
     };
+    if (map.isNotEmpty) {
```

Comment:
I think this function would look cleaner iterating over `map` instead of `environmentMap` and update `environmentMap` accordingly. Something like this
```
map.forEach((String key, String value) {
  if (environmentMap.containsKey(key)) {
    globals.printWarning('...');
  } else {
    // System priority...
    envirionmentMap[key] = value;
  }
});
return environmentMap
```

Reply:
Indeed, it is more clear and clear to write this way,👍🏻

Reply:
You still need to update this fn right?